includes/class-llms-generator-courses.php

```diff
@@ -691,4 +691,45 @@ protected function setup_raw_for_clone( $raw ) {
 
 	}
 
+	/**
+	 * Set all metadata for a given post object.
+	 *
+	 * This method will only set metadata for registered LLMS_Post_Model properties.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Post_Model $post An LLMS post object.
+	 * @param array           $raw  Array of raw data.
+	 * @return void
+	 */
+	protected function set_metadata( $post, $raw ) {
+
+		$generated_from_id = $post->get( 'generated_from_id' );
+
+		if ( $generated_from_id ) {
+			$replace_id_props = array(
+				'course_closed_message',
+				'course_opens_message',
+				'enrollment_closed_message',
+				'enrollment_opens_message',
+			);
+
+			$find    = 'id="' . $generated_from_id . '"';
+			$replace = 'id="' . $post->get( 'id' ) . '"';
+
+			/**
+			 * Replace old post ID with new cloned post ID in course/enrollment
+			 * message shortcodes.
+			 */
+			foreach ( $replace_id_props as $key ) {
+				if ( isset( $raw[ $key ] ) ) {
+					$raw[ $key ] = str_replace( $find, $replace, $raw[ $key ] );
+				}
+			}
```

Comment:
(Some) shortcode attributes will work both with and without quotes (and it's possible they'll work with single quotes also). Possibly a `preg_replace()` regex would work to accommodate all potential iterations of the shortcode attributes.